src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -362,8 +360,11 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
             $name = $attributeMetadata->getName();
 
             if (
-                (false === $groups || array_intersect($attributeMetadata->getGroups(), $groups)) &&
-                $this->isAllowedAttribute($classOrObject, $name, null, $context)
+                (
+                    false === $groups
+                    || \in_array('*', $attributeMetadata->getGroups())
```

Comment:
Wouldn't it be faster to just append `*` to `$attributeMetadata->getGroups()`? At least the result of `$attributeMetadata->getGroups()` should be stored in a variable to avoid calling the method two times.

Reply:
I like this idea.

Reply:
Do you mean `$groups` instead of `$attributeMetadata->getGroups()` ?

I do agree that it would be faster with a temporary variable.

Reply:
@nrobinaubertin I guess that @dunglas had in mind to change:
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php#L33
To something like:
```diff
-    public $groups = [];
+    public $groups = ['*'];
```

Then this `in_array` could be removed as it would be caught by old code with just `array_intersect`.